velox/functions/prestosql/tests/StringFunctionsTest.cpp

```diff
@@ -1201,6 +1201,34 @@ TEST_F(StringFunctionsTest, sha512) {
   EXPECT_EQ(std::nullopt, sha512(std::nullopt));
 }
 
+TEST_F(StringFunctionsTest, HmacSha1) {
+  const auto hmacSha1 = [&](std::optional<std::string> arg,
+                            std::optional<std::string> key) {
+    return evaluateOnce<std::string, std::string>(
+        "hmac_sha1(c0, c1)", {arg, key}, {VARBINARY(), VARBINARY()});
+  };
+  // Use python hmac lib results as the expected value.
+  // >>> import hmac
+  // >>> def sha1(data, key):
+  //         print(hmac.new(key, data, digestmod='sha1').hexdigest())
+  // >>> sha1(b"hashme", b"velox")
+  // d49c944625bdde6c47ad93ea63952bfcf16a630a
+  // >>> sha1(b"Infinity", b"velox")
+  // c19b6b753fe4ac28579c7e84d18feb29760a0d1c
+  // >>> sha1(b"", b"velox")
+  // d0569c4a4f3df995b04ec497b12872c4a2f97517
+  EXPECT_EQ(
+      hexToDec("d49c944625bdde6c47ad93ea63952bfcf16a630a"),
+      hmacSha1("hashme", "velox"));
+  EXPECT_EQ(
+      hexToDec("c19b6b753fe4ac28579c7e84d18feb29760a0d1c"),
+      hmacSha1("Infinity", "velox"));
+  EXPECT_EQ(
+      hexToDec("d0569c4a4f3df995b04ec497b12872c4a2f97517"),
+      hmacSha1("", "velox"));
+  EXPECT_EQ(std::nullopt, hmacSha1(std::nullopt, "velox"));
```

Comment:
Can key be null or empty string?

Reply:
It just to check the cover case.
I see the other tests in StringFunctionsTest.cpp have contain null and "".

Reply:
Can we add to test value is empty or null case as well? Thanks!

Reply:
I still don't get it.
There already have "" and null.

```
  EXPECT_EQ(
      hexToDec("d0569c4a4f3df995b04ec497b12872c4a2f97517"),
      hmacSha1("", "velox"));
  EXPECT_EQ(std::nullopt, hmacSha1(std::nullopt, "velox"));
```

Reply:
Sorry I mean
EXPECT_EQ( hexToDec("d0569c4a4f3df995b04ec497b12872c4a2f97517"), hmacSha1("velox", ""));
EXPECT_EQ(std::nullopt, hmacSha1("velox", std::nullopt));

Reply:
Done and fix a test fail due to illegal character

Reply:
Thanks!